### PR TITLE
testall: Write to stdout using buffer whenever available

### PR DESCRIPTION
**Pull request by thopiekar**
Fixes a build error for Python 2.x.

### CODE REVIEW

File: pywin32_testall.py [hunk 1]
```diff
@@ -20,7 +20,10 @@ def run_test(script, cmdline_rest=""):
     popen = subprocess.Popen(cmd, shell=True, cwd=dirname,
                              stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
     data = popen.communicate()[0]
-    sys.stdout.buffer.write(data)
+    if "buffer" in dir(sys.stdout):
```
Comment (mhammond) [inline pywin32_testall.py]: wouldn't `hasattr()` be better here? And can you please remind me why we use .buffer in the first place? :)

Comment (vernondcole) [reply pywin32_testall.py]: The gurus of Python suggest that hasattr() is usually implemented with a try/except, so it is more efficient to put the try/except in your original code. (Easier to Ask Forgiveness than Permission.)
  Having said that, I have thoroughly messed up this PR with my PR #1445 where I removed the `.buffer` and put in Python3 support.  
  If we really do need `.buffer` I suggest:
```
    if sys.version_info > (3,):
        data =  data.decode('latin-1')
    try:
        sys.stdout.buffer.write(data)
    except AttributeError:
        sys.stdout.write(data)
        sys.stdout.flush()
```

Comment (mhammond) [reply pywin32_testall.py:23]: > The gurus of Python suggest that hasattr() is usually implemented with a try/except, so it is more efficient to put the try/except in your original code.

I'm not too bothered with efficiency here - in this scenario I think readability trumps it. It's also unclear as written - do you expect 'stdout', 'buffer', or 'write' to be the missing attribute? Or possibly all 3? (That's a somewhat rhetorical question obviously :)

@thopiekar, did you plan on revising this?

**Comment by mhammond** [comment]
12 months since the last comment, so I'm closing this - feel free to reopen though if you want to get back to it.

**Pull request closed without merging**